### PR TITLE
ENG-714: limiting-network-player-can-mint-nft

### PR DESCRIPTION
**Pull request by stormcloud266**
add available chains to rewardables. let's us restrict the chains on which an NFT can be minted at the Rewardable level

### CODE REVIEW

File: web/src/components/ClaimNftCell/ClaimNftCell.tsx [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+  rewardable,
+}: CellSuccessProps<FindClaimNftQuery, FindClaimNftQueryVariables>) => {
+  const { chain } = useNetwork()
+  const { isConnected, address } = useAccount()
+  const { openConnectModal } = useConnectModal()
+  const { openChainModal } = useChainModal()
+  const isValidChain = isValidAvailableChain(
+    chain?.id,
+    rewardable.availableChains
+  )
```
Comment (stormcloud266) [inline web/src/components/ClaimNftCell/ClaimNftCell.tsx:91]: This cell is mostly the code from the old ClaimPage, the only big change is it uses this new validChain lookup

File: api/src/graphql/rewardables.sdl.ts [hunk 2]
```diff
@@ -27,6 +28,13 @@ export const schema = gql`
     BUNDLE
   }
 
+  enum AvailableChains {
+    AVAX
+    ETH
+    POLY
+    OPT
+  }
+
```
Comment (illepic) [inline api/src/graphql/rewardables.sdl.ts:37]: Are we forced to dupe this? Or was this autogenerated from the `schema.prisma`?

File: web/src/lib/availableChains.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+} = {
+  [ETH_CHAIN_ID]: 'ETH',
+  [POLYGON_CHAIN_ID]: 'POLY',
+  [AVAX_CHAIN_ID]: 'AVAX',
+  [OPTIMISM_CHAIN_ID]: 'OPT',
+}
+
+export const isValidAvailableChain = (
+  chain: number | undefined,
+  availableChains: AvailableChains[]
```
Comment (illepic) [inline web/src/lib/availableChains.ts:20]: Do you want to send in all available chains every time you call this function?

File: api/db/schema.prisma [hunk 4]
```diff
@@ -42,6 +49,7 @@ model Rewardable {
 
   /// The reward for completion
   nfts Nft[]
+  availableChains AvailableChains[] @default([AVAX, ETH, POLY, OPT])
```
Comment (illepic) [inline api/db/schema.prisma:54]: 1. Use triple slash to explain what this is.
2. Don't you want this `availableChain`s on the NFT model instead?

**Review by illepic — CHANGES_REQUESTED** [review]

File: web/src/lib/availableChains.ts (hunk 3 above)
Comment (stormcloud266) [reply web/src/lib/availableChains.ts:20]: Yes because that could change for each Rewardable

File: api/db/schema.prisma (hunk 4 above)
Comment (stormcloud266) [reply api/db/schema.prisma:54]: I talked to them and they want the Rewardable to set which chains the NFTs can be claimed. If a Rewardable has many NFTs they should all be restricted to the same networks. And they said there might be a case where one NFT is rewarded by multiple Rewardables and the allowed chains might be different on each. 

(update the comment)

File: api/src/graphql/rewardables.sdl.ts (hunk 2 above)
Comment (stormcloud266) [reply api/src/graphql/rewardables.sdl.ts:37]: This was not autogenerated but I don't think it's duplicated anywhere. They basically want to be able to restrict which networks certain NFTs can be claimed. Like the TexturePunx people kept claim on Poly when they should have claimed on Eth, and there was no way to stop it currently

File: api/db/schema.prisma (hunk 4 above)
Comment (illepic) [reply api/db/schema.prisma:54]: Aight, this works. But if we need to store more info about where NFTs connect with Rewardables, we're at the point we'll need a join table (`RewardableNft` or something).

**Review by illepic — APPROVED** [review]

**Pull request merged**